event_model/documents/generate/typeddict_to_schema.py

```diff
@@ -0,0 +1,530 @@
+        origin = get_origin(field_type)
+        args = get_args(field_type)
+        if origin and args:
+            field_type = origin[
+                tuple(
+                    change_sub_typed_dicts_to_basemodels(arg, new_basemodel_classes)
+                    for arg in args
+                )
+            ]
+    return field_type
+
+
+def field_parser(
+    field_name: str, field_type: type, new_basemodel_classes: Dict[str, BaseModel]
+):
+    """Parses a field annotation and generates a pydantic friendly one by
+    extracting relevant information.
+
+    Parameters
+    ----------------
+    field_name : str
+        Name of the field being parsed.
+    field_type : type
+        Annotation to be parsed.
+    new_basemodel_classes : dict
+        Dict where keys are names of _TypedDictMeta classes referenced inside
+        the TypedDict being converted to a BaseModel, and values are BaseModel
+        classes that have already been generated from those TypedDicts -
+        pydantic will throw an error if we try to generate the same BaseModel
+        twice so we just reference the one already created if multiple
+        annotations have the same TypedDict.
+
+    Returns
+    -----------------
+    field_info : type
+        The field_type inputted, but parsed to be pydantic friendly
+    """
+    field_type, is_not_required = field_is_not_required(
+        field_type, remove_origin_if_NotRequired=True
+    )
+
+    as_ref, field_info = get_annotation_contents(field_type)
+
+    field_type = get_field_type(
+        field_name,
+        field_type,
+        new_basemodel_classes,
+    )
+
+    if as_ref:
+        field_type = parse_AsRef(field_type, field_info, as_ref, new_basemodel_classes)
+        field_info.regex = None
+
+    if is_not_required:
+        field_type = Optional[field_type]
+
+    return Annotated[field_type, field_info]
+
+
+class Config(BaseConfig):
+    """Config for generated BaseModel."""
+
+    def alias_generator(string_to_be_aliased):
+        """Alias in snake case"""
+        return re.sub(r"(?<!^)(?=[A-Z])", "_", string_to_be_aliased).lower()
+
+
+def strip_newline_literal(schema: dict):
+    """
+    Pydantic formats the docstring newlines by literally including `\n` in the
+    description. To avoid this uglyness this function swaps `\n` for ` ` in
+    every "description` key in the schema.
+    """
+    for key in schema:
+        if isinstance(schema[key], dict):
+            schema[key] = strip_newline_literal(schema[key])
+        elif key == "description":
+            schema[key] = schema[key].replace("\n", " ")
+    return schema
+
+
+def sort_jsonschema(schema: dict) -> dict:
+    """Sorts the schema properties keys alphabetically by key name, exchanging the
+    properties dicts for OrderedDicts"""
+    for key in schema:
+        if key == "properties":
+            schema[key] = OrderedDict(
+                sorted(list(schema[key].items()), key=lambda x: x[0])
+            )
```

Comment:
Nit: As of python 3.6 all dictionaries keep insertion order, so technically you could get away with a standard dict, but this is probably more readable

Reply:
Yep the alternative with insertions would still require a sort on the keys, I prefer this